### DISCUSSION TITLE
Cant use my LoRA on colab,  getting Error while deserializing header: MetadataIncompleteBuffer (tested with multiple loras)

### DISCUSSION THREAD

**Discussion by nkls2000** [lllyasviel/Fooocus#2591] [Q&A]
Hey! I´ve been training my own model, and I have trained them with both Autotrain (https://colab.research.google.com/github/huggingface/autotrain-advanced/blob/main/colabs/AutoTrain.ipynb)

and with this one:
https://colab.research.google.com/github/huggingface/notebooks/blob/main/diffusers/SDXL_DreamBooth_LoRA_.ipynb

The training went fine, and my model works when I test them on Hugging face. 
I now want to run my own models in Foocus, and I´m trying in the colab version here:
https://colab.research.google.com/github/lllyasviel/Fooocus/blob/main/fooocus_colab.ipynb

I´m getting an error, and its the same error with all my models. It says 
ValueError: Error while deserializing header: MetadataIncompleteBuffer
and then it says my model is corrupt. Would be really happy if anyone could help me with this. How do I solve this?

**Comment by nkls2000** [comment]
Here is the full log:

`[Parameters] Adaptive CFG = 7
[Parameters] Sharpness = 2
[Parameters] ControlNet Softness = 0.25
[Parameters] ADM Scale = 1.5 : 0.8 : 0.3
[Parameters] CFG = 4.0
[Parameters] Seed = 3308499627304362711
[Parameters] Sampler = dpmpp_2m_sde_gpu - karras
[Parameters] Steps = 30 - 15
[Fooocus] Initializing ...
[Fooocus] Loading models ...
Refiner unloaded.
Request to load LoRAs [['aida-robot.safetensors', 0.99], ['None', 1.0], ['None', 1.0], ['None', 1.0], ['None', 1.0]] for model [/content/Fooocus/models/checkpoints/juggernautXL_v8Rundiffusion.safetensors].
Traceback (most recent call last):
  File "/content/Fooocus/modules/patch.py", line 465, in loader
    result = original_loader(*args, **kwargs)
  File "/usr/local/lib/python3.10/dist-packages/safetensors/torch.py", line 259, in load_file
    with safe_open(filename, framework="pt", device=device) as f:
safetensors_rust.SafetensorError: Error while deserializing header: MetadataIncompleteBuffer
 
During handling of the above exception, another exception occurred:
 
Traceback (most recent call last):
  File "/content/Fooocus/modules/async_worker.py", line 913, in worker
    handler(task)
  File "/usr/local/lib/python3.10/dist-packages/torch/utils/_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
  File "/usr/local/lib/python3.10/dist-packages/torch/utils/_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
  File "/content/Fooocus/modules/async_worker.py", line 429, in handler
    pipeline.refresh_everything(refiner_model_name=refiner_model_name, base_model_name=base_model_name,
  File "/usr/local/lib/python3.10/dist-packages/torch/utils/_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
  File "/usr/local/lib/python3.10/dist-packages/torch/utils/_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
  File "/content/Fooocus/modules/default_pipeline.py", line 236, in refresh_everything
    refresh_loras(loras, base_model_additional_loras=base_model_additional_loras)
  File "/usr/local/lib/python3.10/dist-packages/torch/utils/_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
  File "/usr/local/lib/python3.10/dist-packages/torch/utils/_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
  File "/content/Fooocus/modules/default_pipeline.py", line 134, in refresh_loras
    model_base.refresh_loras(loras + base_model_additional_loras)
  File "/usr/local/lib/python3.10/dist-packages/torch/utils/_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
  File "/usr/local/lib/python3.10/dist-packages/torch/utils/_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
  File "/content/Fooocus/modules/core.py", line 95, in refresh_loras
    lora_unmatch = ldm_patched.modules.utils.load_torch_file(lora_filename, safe_load=False)
  File "/content/Fooocus/ldm_patched/modules/utils.py", line 13, in load_torch_file
    sd = safetensors.torch.load_file(ckpt, device=device.type)
  File "/content/Fooocus/modules/patch.py", line 481, in loader
    raise ValueError(exp)
ValueError: Error while deserializing header: MetadataIncompleteBuffer
File corrupted: /content/Fooocus/models/loras/aida-robot.safetensors 
Fooocus has tried to move the corrupted file to /content/Fooocus/models/loras/aida-robot.safetensors.corrupted 
You may try again now and Fooocus will download models again. 
 
Total time: 0.05 seconds`

**Comment by nkls2000** [comment]
here is a url to our model as well:
https://www.dropbox.com/scl/fi/mkuwpedsxbrapt8128uf5/aida-robot.safetensors?rlkey=4otq83i4z1hd72rwuvgrvzg19&raw=1

Model description
LoRA adaption weights for stabilityai/stable-diffusion-xl-base-1.0.

The weights were trained using [DreamBooth](https://dreambooth.github.io/).

LoRA for the text encoder was enabled: False.

Special VAE used for training: madebyollin/sdxl-vae-fp16-fix.

Trigger words
You should use a photo of TOK Aida-robot to trigger the image generation.

**Comment by mashb1t** [comment]
@nkls2000 Relates to https://github.com/AUTOMATIC1111/stable-diffusion-webui/issues/10199.
Which algorithm / settigns have you used to train the LoRA/LyCORIS/LoCon and is there metadata included?

**Reply by nkls2000** [reply]
I used this colab, with the default settings:
https://colab.research.google.com/github/huggingface/notebooks/blob/main/diffusers/SDXL_DreamBooth_LoRA_.ipynb?authuser=1
(as mentioned above I tried the autotrain one too with default settings, same problem)

**Reply by mashb1t** [reply]
I've personally neither encountered this issue nor used that playbook, it's possible that it does not produce correct LoRAs or is misconfigured somehow. Maybe you can try to download the file again, it's not impossible that the file has been corrupted on download.
Maybe somebody else had that problem before, please consult Google / Reddit / StackOverflow or other AI forums for potential solutions.

**Reply by nkls2000** [reply]
I´ve tried to download it multiple times, but the error is persistant. Do you have any suggestion for another colab where I can train my model and see if that works better?

**Reply by mashb1t** [reply]
Not Colab, but i've used kohya_ss to train various LoRAs. There might be working Colab playbooks out there though, also didn't have a deeper look into the issue.

**Reply by nkls2000** [reply]
we just tried with this one as well:
https://colab.research.google.com/github/hollowstrawberry/kohya-colab/blob/main/Lora_Trainer_XL.ipynb#scrollTo=OglZzI_ujZq-

and we are still getting the same output :/
